olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp

```diff
@@ -320,6 +320,42 @@ VersionedLayerClientImpl::PrefetchTiles(PrefetchTilesRequest request) {
                                                           promise);
 }
 
+client::CancellationToken VersionedLayerClientImpl::GetData(
+    TileRequest tile_request, DataResponseCallback callback) {
+  auto schedule_get_data = [&](DataRequest request,
+                               DataResponseCallback callback) {
+    auto catalog = catalog_;
+    auto layer_id = layer_id_;
+    auto settings = settings_;
+
+    auto data_task = [=](client::CancellationContext context) {
+      return repository::DataRepository::GetVersionedData(
+          std::move(catalog), std::move(layer_id), std::move(request), context,
+          std::move(settings));
+    };
+
+    return AddTask(settings.task_scheduler, pending_requests_,
+                   std::move(data_task), std::move(callback));
+  };
+  DataRequest request;
+  request.WithBillingTag(tile_request.GetBillingTag())
+      .WithFetchOption(tile_request.GetFetchOption())
+      .WithPartitionId(tile_request.GetTileKey().ToHereTile());
+  return ScheduleFetch(std::move(schedule_get_data), std::move(request),
+                       std::move(callback));
+}
```

Comment:
The point of this method is to optimize metadata caching for possible upcoming requests by getting quads 4 levels deep instead of a single partition.

See https://developer.here.com/olp/documentation/data-api/api-reference-query.html#operation/quadTreeIndex

Reply:
This implementation makes method to work temporary, before actual implementation is done https://saeljira.it.here.com/browse/OLPEDGE-1529

Reply:
We cannot merge this method as unfinished as we will probably release v1.3 with the new AuthenticationClient by next week and i dont want to risk having v1.3 with normal behaviou and v1.4 with enhanced behaviour. Then please do first the proper impl in VersionedLayerClientImpl and connect afterwards with public API althougt you can also do it together as the overhead for public API is low.